Packet++/header/TcpReassembly.h

```diff
@@ -72,408 +90,479 @@
+		                                    bool enableBaseBufferClearCondition = true)
+		    : removeConnInfo(removeConnInfo), closedConnectionDelay(closedConnectionDelay),
+		      maxNumToClean(maxNumToClean), maxOutOfOrderFragments(maxOutOfOrderFragments),
+		      enableBaseBufferClearCondition(enableBaseBufferClearCondition)
+		{}
 	};
 
 	/**
-	 * An enum for providing reassembly status for each processed packet
+	 * @class TcpReassembly
+	 * A class containing the TCP reassembly logic. Please refer to the documentation at the top of TcpReassembly.h for
+	 * understanding how to use this class
 	 */
-	enum ReassemblyStatus
+	class TcpReassembly
 	{
+	public:
 		/**
-		 * The processed packet contains valid TCP payload, and its payload is processed by `OnMessageReadyCallback` callback function.
-		 * The packet may be:
-		 * 1. An in-order TCP packet, meaning `packet_sequence == sequence_expected`.
-		 *    Note if there's any buffered out-of-order packet waiting for this packet, their associated callbacks are called in this `reassemblePacket` call.
-		 * 2. An out-of-order TCP packet which satisfy `packet_sequence < sequence_expected && packet_sequence + packet_payload_length > sequence_expected`.
-		 *    Note only the new data (the `[sequence_expected, packet_sequence + packet_payload_length]` part ) is processed by `OnMessageReadyCallback` callback function.
+		 * An enum for connection end reasons
 		 */
-		TcpMessageHandled,
+		enum ConnectionEndReason
+		{
+			/** Connection ended because of FIN or RST packet */
+			TcpReassemblyConnectionClosedByFIN_RST,
+			/** Connection ended manually by the user */
+			TcpReassemblyConnectionClosedManually
+		};
+
 		/**
-		 * The processed packet is an out-of-order TCP packet, meaning `packet_sequence > sequence_expected`. It's buffered so no `OnMessageReadyCallback` callback function is called.
-		 * The callback function for this packet maybe called LATER, under different circumstances:
-		 * 1. When an in-order packet which is right before this packet arrives(case 1 and case 2 described in `TcpMessageHandled` section above).
-		 * 2. When a FIN or RST packet arrives, which will clear the buffered out-of-order packets of this side.
-		 *    If this packet contains "new data", meaning `(packet_sequence <= sequence_expected) && (packet_sequence + packet_payload_length > sequence_expected)`, the new data is processed by `OnMessageReadyCallback` callback.
+		 * An enum for providing reassembly status for each processed packet
 		 */
-		OutOfOrderTcpMessageBuffered,
+		enum ReassemblyStatus
+		{
+			/**
+			 * The processed packet contains valid TCP payload, and its payload is processed by `OnMessageReadyCallback`
+			 * callback function. The packet may be:
+			 * 1. An in-order TCP packet, meaning `packet_sequence == sequence_expected`.
+			 *    Note if there's any buffered out-of-order packet waiting for this packet, their associated callbacks
+			 * are called in this `reassemblePacket` call.
```

Comment:
Let's add 3 extra spaces at the beginning of this line?

Reply:
fixed.